### PR TITLE
0x API Marketing Page

### PR DESCRIPTION
**Pull request by johnrjj**
Ready for initial review. 

This PR introduces a 0x API page to our website. Accessed via `https://0x.org/api` when live. This PR also includes peripheral updates to a couple dependencies and various header/footer/homepage/why page updates around 0x API.

Remaining TODOs:
- Update code example to use better web3 spec (no await, use callbacks)
- Add logos of partnerships on 0x API (waiting on marketing/design assets)
- Little bit more clean up if I have time

### CODE REVIEW

File: ts/components/definition.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ export interface DefinitionProps {
     className?: string;
 }
 
-export const Definition: React.FC<DefinitionProps> = ({ className, ...props }: DefinitionProps) => (
+export const Definition: React.FC<DefinitionProps> = React.memo(({ className, ...props }: DefinitionProps) => (
```
Comment (johnrjj) [inline ts/components/definition.tsx:31]: These were _super_ laggy due to context threading (I did some perf analysis via the react profiler). They can all be safely memoized, and increases site perf a good bit (after initial load)

File: package.json [hunk 2]
```diff
@@ -92,7 +92,7 @@
         "sass-loader": "^7.1.0",
         "semver": "5.5.0",
         "semver-sort": "0.0.4",
-        "styled-components": "^4.1.1",
+        "styled-components": "^5.0.0",
```
Comment (johnrjj) [inline package.json:95]: https://styled-components.com/releases#v5.0.0 (lots of cool stuff + no breaking changes)

File: ts/components/blockIconLink.tsx [hunk 3]
```diff
@@ -20,12 +21,17 @@ interface BaseComponentProps {
...
     public onClick = (): void => {
-        const { linkAction, linkUrl } = this.props;
+        const { linkAction, linkUrl, isExternalLink } = this.props;
 
         if (linkAction) {
             linkAction();
+        } else if (isExternalLink) {
+            window.open(
+                linkUrl,
+                '_blank', // New window
```
Comment (kimpers) [inline ts/components/blockIconLink.tsx:31]: might as well
```suggestion
                '_blank', // New window
                'noopener',
```

File: ts/pages/api.tsx [hunk 4]
```diff
@@ -0,0 +1,830 @@
...
+    const copyButtonText = isCopied ? 'Copied!' : 'Copy to clipboard';
+
+    const handleCopyClick = () => {
+        setIsCopied(true);
+        setTimeout(() => setIsCopied(false), 1000);
+    };
+
+    const [quote, setQuote] = useState<any | undefined>();
+
+    const takerAmount = new BigNumber(100000000); // 100$ in usdc (only 8 digits?)
```
Comment (kimpers) [inline ts/pages/api.tsx]: USDC only has 6 decimals, I was equally surprised when I saw it the first time 😄

File: ts/pages/api.tsx [hunk 5]
```diff
@@ -0,0 +1,830 @@
...
+    const formattedTakerAmount = takerAmount // 100$ in usdc (only 8 digits?)
+        .dividedToIntegerBy(1)
+        .toString();
+    const quoteEndpoint = `https://api.0x.org/swap/v0/quote?sellAmount=${formattedTakerAmount}&buyToken=DAI&sellToken=USDC`;
```
Comment (kimpers) [inline ts/pages/api.tsx]: Not sure if the hard code was intentional but we have `utils.getAPIBaseUrl(networkId)` that will return an environment specific API base url.

File: ts/pages/api.tsx [hunk 6]
```diff
@@ -0,0 +1,830 @@
...
+
+    React.useEffect(() => {
+        const fetchQuote = async () => {
+            try {
+                const res = await fetch(quoteEndpoint);
+                const fetchedQuote = await res.json();
+                setQuote(fetchedQuote);
+            } catch (e) {
+                // tslint:disable-next-line: no-console
+                console.log(e);
```
Comment (kimpers) [inline ts/pages/api.tsx]: probably something we want to forward to rollbar, also `@0x/utils` has `logUtils` which allow you to log without messing with linting.

File: ts/pages/api.tsx [hunk 7]
```diff
@@ -0,0 +1,830 @@
...
+                        <AnimationLoader name={'0xApi'} />
+                    </AnimationContainer>
+                }
+                figureMaxWidth="500px"
+                maxWidth="600px"
+                actions={
+                    <Button to={WebsitePaths.DocsApi} isInline={true}>
+                        View API Docs
+                    </Button>
+                    // tslint:disable-next-line:jsx-curly-spacing
```
Comment (kimpers) [inline ts/pages/api.tsx:90]: Unrelated but I vote we just remove this rule discrepancy

File: ts/pages/api.tsx [hunk 8]
```diff
@@ -0,0 +1,830 @@
...
+                            <SyntaxHighlighterContainer>
+                                <SyntaxHighlighter
+                                    style={dark}
+                                    PreTag={TypescriptCodeSamplePre}
+                                    showLineNumbers={true}
+                                    language={'typescript'}
+                                >
+                                    {`// Get a quote to sell 1 ETH to buy DAI
+const response = await fetch(
+  'https://api.0x.org/swap/v0/quote?sellToken=ETH&buyToken=DAI&sellAmount=1000000000000000000'
```
Comment (kimpers) [inline ts/pages/api.tsx:288]: Feel free to ignore this but 1 ETH might be a bit much if people try copy pasting this and accept the tx without realizing that they are making a live trade 😄

File: ts/utils/format_number.ts [hunk 9]
```diff
@@ -366,3 +367,22 @@ export function formatZrx(num: NumStrBigNumber, opts: FormattedCryptoOptions = {
         ...restOpts,
     });
 }
+
+export function formatDai(num: NumStrBigNumber, opts: FormattedCryptoOptions = {}): FormattedNumber {
+    const { fromBaseUnits, ...restOpts } = opts;
+    const normalizedNum: NumStrBigNumber = fromBaseUnits
+        ? Web3Wrapper.toUnitAmount(createBigNumber(num), constants.DECIMAL_PLACES_ZRX)
```
Comment (kimpers) [inline ts/utils/format_number.ts]: NIT: I'd prefer to have a new constant for `DECIMAL_PLACES_DAI` even though it has the same number as `ZRX` feel clearer that way

**Review by kimpers — APPROVED** [review]
Awesome work on this 🚀 🎉  Love the memoization pass!

File: ts/pages/api.tsx (hunk 8 above)
Comment (johnrjj) [reply ts/pages/api.tsx:288]: Agreed - going to touch this up today

File: ts/utils/format_number.ts (hunk 9 above)
Comment (johnrjj) [reply ts/utils/format_number.ts]: Definitely a mistake! Thanks for the callout :) Fixed

File: ts/pages/api.tsx (hunk 6 above)
Comment (johnrjj) [reply ts/pages/api.tsx]: Ah, thanks, will do a quick console/logUtils pass

File: ts/pages/api.tsx (hunk 5 above)
Comment (johnrjj) [reply ts/pages/api.tsx]: Fixed! Thanks

File: ts/pages/api.tsx (hunk 4 above)
Comment (johnrjj) [reply ts/pages/api.tsx]: So strange haha, I updated the comment to reflect it has 6 digits and we're making a 100$ trade

**Pull request merged**